src/server.ts

```diff
@@ -235,52 +238,114 @@ app.post("/pdf/authenticate", async (c) => {
   }
 });
 
+// PDF Session Authentication Status Check Route
+app.get("/pdf/is-session-authenticated", async (c) => {
+  try {
+    const sessionId = c.req.query("sessionId");
+
+    if (!sessionId) {
+      return c.json({ error: "sessionId parameter is required" }, 400);
+    }
+
+    // Get the SessionFileTracker Durable Object for this session
+    const sessionIdObj = c.env.SessionFileTracker.idFromName(sessionId);
+    const sessionTracker = c.env.SessionFileTracker.get(sessionIdObj);
+
+    const authCheckResponse = await sessionTracker.fetch(
+      "https://dummy-host/is-session-authenticated",
```

Comment:
cool URL bruh. also (for another PR, can be mine), would be better in a config file